packages/registry/src/RegistryFactory.ts

```diff
@@ -0,0 +1,21 @@
+import { HttpProvider } from './ports/PoapRegistry/HttpProvider';
+import { PoapRegistryProvider } from './ports/PoapRegistry/PoapRegistryProvider';
+import { RegisrtyProvider } from './ports/RegistryProvider';
+import { Registry } from './Registry';
+
+export class RegistryFactory {
+  createRegistry(provider?: RegisrtyProvider): Registry {
+    if (provider) {
+      return new Registry(provider);
+    } else {
+      throw new Error('Provider is required');
```

Comment:
Do not throw `Error`, better to create an error class extending `Error`. But in this case better do the argument required.